azure-pipelines.yml

```diff
@@ -4,16 +4,368 @@
+- task: PublishTestResults@2
+  inputs:
+    testResultsFiles: '$(System.DefaultWorkingDirectory)/sonic-syseepromd/test-results.xml'
+    testRunTitle: sonic-syseepromd (Python 3)
+    failTaskOnFailedTests: true
+  condition: succeededOrFailed()
+  displayName: 'Publish sonic-syseepromd (Python 3) test results'
+
+- task: PublishCodeCoverageResults@1
+  inputs:
+    codeCoverageTool: Cobertura
+    summaryFileLocation: '$(System.DefaultWorkingDirectory)/sonic-syseepromd/coverage.xml'
+    reportDirectory: '$(System.DefaultWorkingDirectory)/sonic-syseepromd/htmlcov/'
+  displayName: 'Publish sonic-syseepromd (Python 3) test coverage'
+
+- script: |
+    set -e
+    python3 setup.py bdist_wheel
+  workingDirectory: sonic-syseepromd
+  displayName: 'Build sonic-syseepromd (Python 3)'
+
+- publish: '$(System.DefaultWorkingDirectory)/sonic-syseepromd/dist/'
+  artifact: sonic-syseepromd
+  displayName: "Publish sonic-syseepromd wheels"
+
+# sonic-thermalctld
+
+- script: |
+    python2 setup.py test
+  workingDirectory: sonic-thermalctld
+  displayName: 'Test sonic-thermalctld (Python 2)'
+
+- task: PublishTestResults@2
+  inputs:
+    testResultsFiles: '$(System.DefaultWorkingDirectory)/sonic-thermalctld/test-results.xml'
+    testRunTitle: sonic-thermalctld (Python 2)
+    failTaskOnFailedTests: true
+  condition: succeededOrFailed()
+  displayName: 'Publish sonic-thermalctld (Python 2) test results'
+
+- task: PublishCodeCoverageResults@1
+  inputs:
+    codeCoverageTool: Cobertura
+    summaryFileLocation: '$(System.DefaultWorkingDirectory)/sonic-thermalctld/coverage.xml'
+    reportDirectory: '$(System.DefaultWorkingDirectory)/sonic-thermalctld/htmlcov/'
+  displayName: 'Publish sonic-thermalctld (Python 2) test coverage'
+
+- script: |
+    set -e
+    python2 setup.py bdist_wheel
+  workingDirectory: sonic-thermalctld
+  displayName: 'Build sonic-thermalctld (Python 2)'
+
+- script: |
+    python3 setup.py test
+  workingDirectory: sonic-thermalctld
+  displayName: 'Test sonic-thermalctld (Python 3)'
+
+- task: PublishTestResults@2
+  inputs:
+    testResultsFiles: '$(System.DefaultWorkingDirectory)/sonic-thermalctld/test-results.xml'
+    testRunTitle: sonic-thermalctld (Python 3)
+    failTaskOnFailedTests: true
+  condition: succeededOrFailed()
+  displayName: 'Publish sonic-thermalctld (Python 3) test results'
+
+- task: PublishCodeCoverageResults@1
+  inputs:
+    codeCoverageTool: Cobertura
+    summaryFileLocation: '$(System.DefaultWorkingDirectory)/sonic-thermalctld/coverage.xml'
+    reportDirectory: '$(System.DefaultWorkingDirectory)/sonic-thermalctld/htmlcov/'
+  displayName: 'Publish sonic-thermalctld (Python 3) test coverage'
+
+- script: |
+    set -e
+    python3 setup.py bdist_wheel
+  workingDirectory: sonic-thermalctld
+  displayName: 'Build sonic-thermalctld (Python 3)'
+
+- publish: '$(System.DefaultWorkingDirectory)/sonic-thermalctld/dist/'
+  artifact: sonic-thermalctld
+  displayName: "Publish sonic-thermalctld wheels"
```

Comment:
suggest to use template, I can see lots of duplication. check the sonic-mgmt/azure-pipelines.yml

Reply:
@lguohan: I templatized my solution, but it appears that Azure Pipelines runs the templates in the host OS rather than the specified container, so the tasks fail. I can't seem to find anything in the documentation regarding this. Do you have any pointers?